app/src/behaviors/behavior_default_layer.c

```diff
@@ -0,0 +1,176 @@
+#include <zmk/endpoints.h>
+#include <zmk/keymap.h>
+
+#include <zmk/event_manager.h>
+#include <zmk/events/endpoint_changed.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct default_layer_settings_t {
+    uint8_t usb[ZMK_ENDPOINT_USB_COUNT];
+    uint8_t ble[ZMK_ENDPOINT_BLE_COUNT];
+};
+
+static struct default_layer_settings_t default_layers = {0};
+
+static int apply_default_layer_config(struct zmk_endpoint_instance endpoint) {
+    uint8_t layer = 0;
+
+    switch (endpoint.transport) {
+    case ZMK_TRANSPORT_USB:
+        __ASSERT(ZMK_ENDPOINT_USB_COUNT == 1, "Unreachable");
+        layer = default_layers.usb[0];
+        break;
+
+    case ZMK_TRANSPORT_BLE:
+        __ASSERT(endpoint.ble.profile_index < ZMK_ENDPOINT_BLE_COUNT, "Unreachable");
+        layer = default_layers.ble[endpoint.ble.profile_index];
+        break;
+    }
+
+    int ret = zmk_keymap_layer_set_default(layer);
+    if (ret < 0) {
+        LOG_WRN("Could not apply default layer from settings. Perhaps something in the code/keymap "
+                "changed since configuration was saved.");
+        return ret;
+    }
+
+    LOG_INF("Activated default layer (%d) for the current endpoint.", layer);
+    return 0;
+}
+
+static int default_layer_set(const char *name, size_t len, settings_read_cb read_cb, void *cb_arg) {
+    const char *next;
+    int rc;
+
+    if (settings_name_steq(name, "settings", &next) && !next) {
+        if (len != sizeof(default_layers)) {
+            return -EINVAL;
+        }
+
+        rc = read_cb(cb_arg, &default_layers, sizeof(default_layers));
+        if (rc >= 0) {
+            return 0;
+        }
+
+        return rc;
+    }
+
+    return -ENOENT;
+}
+
+struct settings_handler default_layer_conf = {
+    .name = "default_layer",
+    .h_set = default_layer_set,
+};
+
+static int default_layer_init(void) {
+    settings_subsys_init();
+
+    int ret = settings_register(&default_layer_conf);
+    if (ret) {
+        LOG_ERR("Could not register default layer settings (%d).", ret);
+        return ret;
+    }
+
+    settings_load_subtree("default_layer");
+
+    return apply_default_layer_config(zmk_endpoints_selected());
+}
+SYS_INIT(default_layer_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
+
+static int save_default_layer_setting(uint8_t layer, struct zmk_endpoint_instance endpoint) {
+    if (layer >= ZMK_KEYMAP_LAYERS_LEN) {
+        return -EINVAL;
+    }
+
+    switch (endpoint.transport) {
+    case ZMK_TRANSPORT_USB:
+        __ASSERT(ZMK_ENDPOINT_USB_COUNT == 1, "Unreachable");
+        default_layers.usb[0] = layer;
+        break;
+
+    case ZMK_TRANSPORT_BLE:
+        __ASSERT(endpoint.ble.profile_index < ZMK_ENDPOINT_BLE_COUNT, "Unreachable");
+        default_layers.ble[endpoint.ble.profile_index] = layer;
+        break;
+    }
+
+    int ret = settings_save_one("default_layer/settings", &default_layers, sizeof(default_layers));
```

Comment:
Do we want to apply debouncing here like most other settings that save to flash?